youtube_dl/downloader/external.py

```diff
@@ -27,15 +27,18 @@ def real_download(self, filename, info_dict):
 
         retval = self._call_downloader(tmpfilename, info_dict)
         if retval == 0:
-            fsize = os.path.getsize(encodeFilename(tmpfilename))
-            self.to_screen('\r[%s] Downloaded %s bytes' % (self.get_basename(), fsize))
-            self.try_rename(tmpfilename, filename)
-            self._hook_progress({
-                'downloaded_bytes': fsize,
-                'total_bytes': fsize,
-                'filename': filename,
-                'status': 'finished',
-            })
+            if filename == '-':
+                self._hook_progress({'status': 'finished'})
```

Comment:
I guess, ok, done.
It's not clear what the contact to `self._hook_progress()` really is, and what point filenames are really only actual files in the filesystem, and at what point we maintain the fiction of '-' meaning stdout. But it doesn't look like it matters either way.